flight/Modules/Actuator/actuator.c

```diff
@@ -191,76 +214,34 @@ static void actuatorTask(void* parameters)
 	{
 		PIOS_WDG_UpdateFlag(PIOS_WDG_ACTUATOR);
 
-		// Wait until the ActuatorDesired object is updated
-		bool rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		bool rc=false;
 
-		/* Process settings updated events even in timeout case so we always act on the latest settings */
-		if (actuator_settings_updated) {
-			actuator_settings_updated = false;
-			ActuatorSettingsGet (&actuatorSettings);
-			actuator_update_rate_if_changed (&actuatorSettings, false);
+		// If it's our first iteration, fall down to getting the
+		// config and enabling failsafe.
+		if (!first) {
+			UAVObjEvent ev;
+			// Wait until the ActuatorDesired object is updated
+			rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		} else {
+			first=false;
 		}
-		if (mixer_settings_updated) {
-			mixer_settings_updated = false;
-			MixerSettingsGet (&mixerSettings);
-
-			for (int i=0; i<ACTUATORCOMMAND_CHANNEL_NUMELEM; i++) {
-				switch(i) {
-				case 0:
-					MixerEntry_t[i].type = mixerSettings.Mixer1Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer1Vector;
-					break;
-				case 1:
-					MixerEntry_t[i].type = mixerSettings.Mixer2Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer2Vector;
-					break;
-				case 2:
-					MixerEntry_t[i].type = mixerSettings.Mixer3Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer3Vector;
-					break;
-				case 3:
-					MixerEntry_t[i].type = mixerSettings.Mixer4Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer4Vector;
-					break;
-				case 4:
-					MixerEntry_t[i].type = mixerSettings.Mixer5Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer5Vector;
-					break;
-				case 5:
-					MixerEntry_t[i].type = mixerSettings.Mixer6Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer6Vector;
-					break;
-				case 6:
-					MixerEntry_t[i].type = mixerSettings.Mixer7Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer7Vector;
-					break;
-				case 7:
-					MixerEntry_t[i].type = mixerSettings.Mixer8Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer8Vector;
-					break;
-				case 8:
-					MixerEntry_t[i].type = mixerSettings.Mixer9Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer9Vector;
-					break;
-				case 9:
-					MixerEntry_t[i].type = mixerSettings.Mixer10Type;
-					MixerEntry_t[i].vector = mixerSettings.Mixer10Vector;
-					break;
-				default:
-					// We can never get here unless there are mixer channels not handled in the above. Fail out.
-					PIOS_Assert(0);
-				}
-			}
+
+		/* Process settings updated events even in timeout case so we always act on the latest settings */
```

Comment:
I don't know, it's what my fingers did.  I never know how long a comment is going to be until I'm done typing ;)